server/controllers/user.js

```diff
@@ -19,5 +19,22 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
     throw new Error("No users found in search");
   }
 
-  res.status(200).json({ users: users });
+  res.status(200).json({ users });
 });
+
+// @route POST /load
+// @desc load user profile data for one or many users
+// @access Private
+exports.loadUsersData = async (req, res) => {
+  const { users } = req.body;
+
+  const profiles = await User.find({ _id: { $in: users } })
+    .select("profile")
+    .populate("profile", "firstName lastName photo");
+
+  if (!profiles) {
+    res.status(400).send("An error occurred while retrieving profile data.");
```

Comment:
```suggestion
    res.status(500).send("An error occurred while retrieving profile data.");
```
This will be a server error here